plugin/rename_file.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import annotations
+from .core.open import open_file_uri
+from .core.promise import Promise
+from .core.protocol import Notification, Request
+from .core.registry import LspWindowCommand
+from .core.types import match_file_operation_filters
+from .core.url import filename_to_uri
+from functools import partial
+from pathlib import Path
+from typing import TYPE_CHECKING
+import sublime
+import sublime_plugin
+import weakref
+
+if TYPE_CHECKING:
+    from ..protocol import FileRename
+    from ..protocol import WorkspaceEdit
+    from .core.sessions import Session
+    from collections.abc import Generator
+
+
+class LspRenameFromSidebarOverrideCommand(LspWindowCommand):
+    def is_enabled(self) -> bool:
+        return True
+
+    def run(self, paths: list[str] | None = None) -> None:
+        old_path = paths[0] if paths else None
+        if old_path:
+            self.window.run_command('lsp_rename_path', {
+                "old_path": old_path
+            })
```

Comment:
I have not followed the recent discussions, so sorry if this has been mentioned before; The only thing what this command does is to call another command. And both are WindowCommands. So couldn't they be combined into a single command? The `paths` parameter has a default value, so I think you could just add it to the other command and then use
```python
if paths:  # command was called from side bar context menu
    old_path = paths[0]
```

Reply:
[b27aa61](https://github.com/sublimelsp/LSP/pull/2498/commits/b27aa61f6e3fdfffa109309409b3fecfbb97fc0c)